test/controllers/contactPreference/ChangeEmailControllerSpec.scala

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.contactPreference
+
+import assets.CustomerInfoConstants.fullCustomerInfoModel
+import common.SessionKeys
+import connectors.httpParsers.GetCustomerInfoHttpParser.GetCustomerInfoResponse
+import controllers.ControllerBaseSpec
+import models.errors.ErrorModel
+import org.jsoup.Jsoup
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{never, verify, when}
+import play.api.http.Status
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.email.CaptureEmailView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ChangeEmailControllerSpec extends ControllerBaseSpec {
+
+  val testValidationEmail: String = "validation@example.com"
+  val testValidEmail: String = "pepsimac@gmail.com"
+  val testInvalidEmail: String = "invalidEmail"
+  val view: CaptureEmailView = injector.instanceOf[CaptureEmailView]
+
+  def setup(result: GetCustomerInfoResponse): Any =
+    when(mockVatSubscriptionService.getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext]))
```

Comment:
We already have an existing mocking function for this service (in `MockVatSubscriptionService`) that uses ScalaMock instead of Mockito. For consistent mocking results it's probably best to reuse that instead of defining your own `when().thenReturn()` function here.